### ISSUE TITLE
Intermittent h2 errors on latest version

### ISSUE THREAD

**Issue by rotate4all** [icing/mod_h2#226]
Ever since updating to the latest Apache and mod_http2 version, I've been intermittently getting the following errors logged in apache log:
`http2:error] [pid 784963:tid 47693616867072] (70014)End of file found: [client X.X.X.X:XXXXX] AH10021: h2_mplx(XXX): register at workers`

There are about 400 of these logged so far since October 27 (when I did the update) but even so it's frustrating and I would certainly like to track down the source of them.

Running on the following:
```
mod_http2 (v1.15.24, feats=CHPRIO+SHA256+INVHD+DWINS, nghttp2 1.44.0)
Apache/2.4.51 (cPanel) OpenSSL/1.1.1l
```
So what do these "End of file found" errors apparently caused by "h2_mplx" mean anyway?
Any pointers would be appreciated.
Thanks.

**Comment by icing** [comment]
We are currently investigating https://github.com/icing/mod_h2/issues/212 where these errors are also logged and I just proposed a patch for testing.

The reason these keep appearing, my current guess, is the change in handling of graceful restarts. In 2.4.51, when a graceful is done, h2 shuts down its idle worker threads and will should longer accept new requests from the client. This seems not to work correctly and starting these unwanted requests gives that error.

**Comment by rotate4all** [comment]
Thanks for the reply.
I rarely restart my Apache. I did do a couple of graceful restarts manually after the said errors started happening but that haven't made any difference.
I am using the following directive in Apache config since ages ago:
`MaxConnectionsPerChild  15000`
So in theory after reaching that many requests (sometimes quite often as it's a busy server), the child request dies and a new one gets spawned. This I suspect may be affected by the issue you mentioned as I assume it uses the same graceful restart mechanism as the main server.
Just adding more context. Hopefully it gets you to the root cause.

**Comment by icing** [comment]
I just release v1.15.25 with a fix for #212. This should solve your error log messages as well.
